### PR TITLE
feat(complex-example): refactor with makeMock utility

### PR DESCRIPTION
**Pull request by WonderPanda**

### CODE REVIEW

File: complex-sample/src/cat/cat.guard.spec.ts [hunk 1]
```diff
@@ -11,29 +13,30 @@ describe('CatGuard', () => {
...
   it('should return true with auth', () => {
-    expect(
-      guard.canActivate({
-        switchToHttp: () => ({
-          getRequest: () => ({
-            headers: {
-              authorization: 'auth',
-            },
-          }),
+    const context = makeMock<ExecutionContext>({
```
Comment (jmcdo29) [inline complex-sample/src/cat/cat.guard.spec.ts]: Can this now be done with even less mocking as you were showing me on discord? Something like
```ts
const context = makeMock<ExecutionContext>();
context.switchToHttp().getRequest = mockImplementation(() => ({
  headers: { authorization: "auth" }
}));
```

File: complex-sample/src/cat/cat.guard.spec.ts [hunk 2]
```diff
@@ -11,29 +13,30 @@ describe('CatGuard', () => {
...
   it('should return false without auth', () => {
-    expect(
-      guard.canActivate({
-        switchToHttp: () => ({
-          getRequest: () => ({
-            headers: {
-              authorization: '',
-            },
-          }),
+    const context = makeMock<ExecutionContext>({
```
Comment (jmcdo29) [inline complex-sample/src/cat/cat.guard.spec.ts]: I think I was being verbose with adding the authorization header. We can probably just use `const context = makeMock<ExecutionContext>()` and see it come back falsy.

Comment (WonderPanda) [reply complex-sample/src/cat/cat.guard.spec.ts]: Oh nice! I didn't even think of that, great call and another awesome use case for this library

**Comment by jmcdo29** [comment]
I'm looking forward to the package being released! I think it's gonna be helpful for a ton of people! I'll get some documentation written up for it soon so that others can understand it as well

**Pull request merged**